content/blog/2018-10-09-jstor.md

```diff
@@ -0,0 +1,512 @@
+parts,
+I was able to trim down execution time considerably overall. For 25,000 files,
+which is the maximum number of files you can receive at one time through the
+standard interface of JSTOR/DfR, execution time is slightly under 4 minutes, or
+2 minutes if executed in parallel (at least on my machine with a 2,8GHz Intel 
+i5).
+
+
+### Lessons learned
+I have learned many things while working on this package. While I acquired 
+certain skills (like writing simple XPATH-queries), I want to highlight a few 
+general things.
+
+Something which is probably true for many people working with and developing for
+R, is that community is important. Without the efforts of many others, through
+paid work or by spending their time voluntarily, developing the package would
+not have been possible. This is true for the many packages my code builds on,
+for the suite of packages that helps in developing and maintaining packages,
+for answers I got over StackOverflow, and last but not least for the helpful
+comments I received by @elinw and @jsonbecker during the review process for
+onboarding the package.
+
+A lesson which is more or less obvious from what I wrote above would be to 
+benchmark your code, if you are planning on running it often. Packages like
+`microbenchmark`, `profvis` or `bench` can help you in different ways to make 
+sure that your code runs more efficiently. 
+
+At the beginning of the post I mentioned, that there might be a better
+approach on parsing those files altogether. Before I finish, I want to briefly
+elaborate on that thought.
+
+The general approach I took when writing the package was inspired by the idea of
+functional programming, that can be implemented in R through `lapply` or similar
+functions within the `purrr`-package. The approach is to write a function that
+solves your problem for one case, and then to apply it to all cases. In my case,
+this leads to some duplication and inefficiency: The package has several 
+functions that extract certain parts of the metadata-files. This makes sense,
+since you only might be interested in certain parts, and parsing everything else
+would be a waste of time. Unfortunately, if you happen to be interested in two
+parts all the time, this means that each function has to read the original file
+separately. There is no economy of scales, so to speak, because the file is
+read again for each part which is to be extracted. An alternative would have 
+been to write a general wrapper function that reads in the files and then 
+applies each function in turn. I suspect that this would be quite complex, given
+that the code should also be able to run in parallel in a proper way. This, and
+the fact that I already spent *a lot* of time writing the package, mean that
+I will most likely not add improvements on this side. There are however many 
+other options for improvement, and I will gladly point you to a few of them.
+
+### Options to contribute
```

Comment:
This is great as a last section!
If you label them `hacktoberfest` like these: https://github.com/search?q=user%3Aropensci+label%3Ahacktoberfest then you might get more eyes on them in October

Context: https://twitter.com/rOpenSci/status/1044708122265350145

Reply:
Thanks for mentioning this!